### ISSUE TITLE
FR: warn about implicit 0 offset with bash substring expansion (${VAR::-2})

### ISSUE THREAD

**Issue by vapier** [koalaman/shellcheck#2633]
#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
echo "${USER::2}"
```

#### Here's what shellcheck currently says:

nothing

#### Here's what I wanted or expected to see:

it should recommend using 0 as an explicit offset.  it's unclear whether the author typoed and meant `${USER:2}`.

other things to consider: if they wanted to use a negative value, the 0 would often show up anyways since `${USER:-2}` is not the same as `${USER:0-2}`.
